05.wc/lib/wc.rb

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+
+def wc_output
+  opt = OptionParser.new
+  wc_options = { lines: false, words: false, bytes: false }
+  opt.on('-l') { |v| wc_options[:lines] = v }
+  opt.on('-w') { |v| wc_options[:words] = v }
+  opt.on('-c') { |v| wc_options[:bytes] = v }
+  opt.parse!(ARGV)
+
+  counts_by_file = read_files
+  total_by_file = format_file(wc_options, counts_by_file)
+  total_of_files = format_total(wc_options, counts_by_file) if counts_by_file.size > 1
+  [total_by_file, total_of_files].join
+end
+
+def read_files
+  counts = []
+  l = 0
+  w = 0
+  c = 0
+  ARGF.each do |line|
+    l += 1
+    w += line.split.size
+    c += line.bytesize
+    next unless ARGF.file.eof?
+
+    counts << { lines: l, words: w, bytes: c, path: $stdin.tty? ? ARGF.file.path : nil }
+    l = 0
+    w = 0
+    c = 0
+  end
+  counts
+end
+
+def format_file(wc_options, counts)
```

Comment:
引数で渡されているのはfileではなくcountsなので、フォーマットしてるのはfile?counts?どっち？？と思いました。
あと、慣習的にoptionは後ろに持ってくることが多いですね。
```suggestion
def format_file(counts, wc_options)
```

Reply:
メソッド名を`format_counts`に変更しました。
慣習についても教えていただき、ありがとうございます。引数の位置も変更しました。